interni/norme-di-progetto/components/processi-primari/sviluppo.tex

```diff
@@ -0,0 +1,678 @@
+Essendo l'architettura a garantire l'organizzazione e la coesione dei componenti si intuisce come rivesta un ruolo fondamentale per ottenere un prodotto di qualità.
+A tal proposito essa dovrà avere le seguenti caratteristiche:
+\begin{description}
+  \item [Sufficienza]: deve soddisfare tutti i requisiti definiti nel documento \textit{Analisi dei requisiti (versione \versione)}.
+  \item [Comprensibilità]: deve essere comprensibile a tutti gli stakeholders.
+  \item [Modularità]: deve essere suddivisa in parti chiare e distinte minimizzando le dipendenze.
+  \item [Robustezza]: deve saper tollerare carichi di lavoro in ingresso variabili.
+  \item [Flessibilità]: al variare dei requisiti deve essere possibile apportare modifiche senza problemi.
+  \item [Riusabilità]: le sue parti possono essere reimpiegabili.
+  \item [Efficienza]: deve fare uso di un adeguato numero di risorse.
+  \item [Affidabilità]: deve svolgere il compito per il quale è stata realizzata.
+  \item [Disponibilità]: deve essere disponibile.
+  \item [Sicurezza rispetto a malfunzionamenti]: non deve avere malfunzionamenti gravi.
+  \item [Sicurezza rispetto a intrusioni]: deve tutelare ogni forma di dato sensibile e non deve essere vulnerabile.
+  \item [Semplicità]: deve contenere tutto il necessario e nulla di superfluo.
+  \item [Incapsulazione]: le componenti non devono essere visibili all'esterno.
+  \item [Coesione]: le parti devono condividere i medesimi obiettivi.
+  \item [Basso accoppiamento]: deve minimizzare le dipendenze.
+\end{description}
+
+\subparagraph{Diagrammi UML}%
+\label{subp:diagrammi_UML}
+Per illustrare le proprie scelte progettuali GruppOne realizzerà dei diagrammi mediante il linguaggio visivo \glossario{UML 2.0}.
+Tali diagrammi includono:
+\begin{itemize}
+  \item Diagrammi delle classi: presentano i metodi, gli attributi e le relazioni di ciascun oggetto con il sistema.
+  \item Diagrammi di package: illustrano come sono raggruppate le classi che condividono elementi comuni mediante un'unità di livello più alto.
+  \item Diagrammi di attività: descrivono la logica procedurale dei casi d'uso e ne mostrano gli aspetti dinamici.
+  \item Diagrammi di sequenza: definiscono come gli oggetti collaborano tra di loro per riprodurre un comportamento comune.
+\end{itemize}
+Nei prossimi paragrafi verranno presentate le norme a cui i progettisti dovranno attenersi nella produzione dei diagrammi.
+
+\subparagraph{Diagrammi delle classi}%
+\label{subp:diagrammi_delle_classi}
+Una classe, in UML, ha la sottostante rappresentazione:
+\begin{figure}[H]%
+  \label{fig:classe_uml}
+  \includegraphics[width=3cm]{class-example.png}
+  \centering
+  \caption{diagramma delle classi realizzato con PlantUML}
+\end{figure}
+\begin{description}
+  \item [Nomi e visibilità]: i nomi di classi, campi dati e metodi andranno indicati in minuscolo e scritti in lingua inglese.
```

Comment:
"campi dati e metodi" -> "i campi dati e i metodi"

Reply:
questo mi sembra corretto già così, perché c'è il "i nomi di" davanti

Reply:
ho sistemato in "i nomi delle classi, dei campi dati e dei metodi ..."

Reply:
Ah hai ragione, in realtà andava bene anche originalmente. A posto